packages/json-parser-products/test/mappings.spec.js

```diff
@@ -0,0 +1,620 @@
+          images: oneLineTrim`
+            https://example.com/foobar/commer.jpg|3|4;
+            https://example-2.com/demo/tools.jpg|1|5|image-label`,
+        },
+        state: 'my-resolved-state',
+        hasStagedChanges: false,
+      }
+      expect(productMapping._mapProperties(sample)).toEqual(expected)
+    })
+  })
+
+  describe('::mapCategories', () => {
+    it('resolves categories to string by name', () => {
+      const sampleCat = [{
+        name: {
+          en: 'cat-in-en',
+          de: 'cat-in-de',
+        },
+        externalId: 'cat-ext-id',
+      }, {
+        name: {
+          en: 'cat-foo-en',
+          de: 'cat-foo-de',
+        },
+        externalId: 'cat-foo-id',
+      }]
+      const expected = 'cat-in-en;cat-foo-en'
+      const args = ['name', ';', 'en']
+      expect(ProductMapping._mapCategories(sampleCat, ...args)).toBe(expected)
+    })
+
+    it('resolves categories to string by externalId', () => {
+      const sampleCat = [{
+        name: {
+          en: 'cat-in-en',
+          de: 'cat-in-de',
+        },
+        externalId: 'cat-ext-id',
+      }, {
+        name: {
+          en: 'cat-foo-en',
+          de: 'cat-foo-de',
+        },
+        externalId: 'cat-foo-id',
+      }]
+      const expected = 'cat-ext-id;cat-foo-id'
+      const args = ['externalId', ';']
+      expect(ProductMapping._mapCategories(sampleCat, ...args)).toBe(expected)
+    })
+
+    it('resolves categories to string by key', () => {
+      const sampleCat = [{
+        name: {
+          en: 'cat-in-en',
+          de: 'cat-in-de',
+        },
+        key: 'cat-key',
+      }, {
+        name: {
+          en: 'cat-foo-en',
+          de: 'cat-foo-de',
+        },
+        key: 'cat-keysss',
+      }]
+      const expected = 'cat-key;cat-keysss'
+      const args = ['key', ';']
+      expect(ProductMapping._mapCategories(sampleCat, ...args)).toBe(expected)
+    })
+
+    it('resolves categories to string by `namedPath`', () => {
+      const sampleCat = [{
+        name: {
+          en: 'cat-in-en',
+          de: 'cat-in-de',
+        },
+        parent: {
+          name: {
+            en: 'parent',
+          },
+          parent: {
+            name: {
+              en: 'grandparent',
+            },
+            parent: {
+              name: {
+                en: 'greatgrandparent',
+              },
+            },
+          },
+        },
+        key: 'cat-key',
+      }, {
+        name: {
+          en: 'cat-foo-en',
+          de: 'cat-foo-de',
+        },
+        parent: {
+          name: {
+            en: 'another-parent',
+          },
+        },
+        key: 'cat-keysss',
+      }]
+      const args = ['namedPath', ';', 'en']
+      const expected = oneLineTrim`
+        greatgrandparent>grandparent>parent>cat-in-en
+        ;another-parent>cat-foo-en`
+      expect(ProductMapping._mapCategories(sampleCat, ...args)).toBe(expected)
+    })
+
+    describe('::retrieveNamedPath', () => {
```

Comment:
Isn't this test similar to the previous one? Can it be merged somehow?

Reply:
It can, but they're testing different methods